pkg/couchdb/mango/index.go

```diff
@@ -26,9 +26,12 @@ type Index struct {
 
 // IndexOnFields constructs a new Index
 // it lets couchdb defaults for index & designdoc names.
-func IndexOnFields(doctype string, fields ...string) *Index {
+func IndexOnFields(doctype, name string, fields []string) *Index {
 	return &Index{
 		Doctype: doctype,
-		Request: &IndexRequest{Index: IndexFields(fields)},
+		Request: &IndexRequest{
+			DDoc:  name,
```

Comment:
Using DDoc value here means index cand be duplicated or broken by developpers mistake.

Example : 
`IndexOnFields("io.cozy.files", "by-mime-type", []string{"mimetype"})` vs `IndexOnFields("io.cozy.files", "by-type", []string{"mimetype"})` vs `IndexOnFields("io.cozy.files", "by-type", []string{"type"})`

While it will probably not occurs as all indexes are defined in consts, why do we need to name design docs?